components/eam/src/chemistry/mozart/mo_drydep.F90

```diff
@@ -2523,723 +520,684 @@ subroutine drydep_xactive( ncdate, sfc_temp, pressure_sfc,  &
-                      endif
-                   end if
-                   if( has_rain(i) ) then
-                      ! rlux(i,lt,o3_ndx) = 1./(1.e-3 + (1./(3.*rlu(sndx,lt))))
-                      rlux_o3(i,lt)     = 3000._r8*rlu(sndx,lt)/(1000._r8 + 3._r8*rlu(sndx,lt))
-                      if( o3_ndx > 0 ) then
-                         rlux(i,lt,o3_ndx) = rlux_o3(i,lt)
-                      endif
-                      if( o3a_ndx > 0 ) then
-                         rlux(i,lt,o3a_ndx) = rlux_o3(i,lt)
-                      endif
-                   end if
-                end if
-
-                if ( o3_ndx > 0 ) then
-                   rclx(i,lt,o3_ndx) = cts(i) + rclo(index_season(i,lt),lt)
-                   rlux(i,lt,o3_ndx) = cts(i) + rlux(i,lt,o3_ndx)
-                end if
-                if ( o3a_ndx > 0 ) then
-                   rclx(i,lt,o3a_ndx) = cts(i) + rclo(index_season(i,lt),lt)
-                   rlux(i,lt,o3a_ndx) = cts(i) + rlux(i,lt,o3a_ndx)
-                end if
+                if( sfc_temp(icol) > tmelt ) then
+                   if( has_dew(icol) ) then
+                      rlux_o3(icol,lt)     = 3000._r8*rlu(sndx,lt)/(1000._r8 + rlu(sndx,lt))
+                   endif
+                   if( has_rain(icol) ) then
+                      rlux_o3(icol,lt)     = 3000._r8*rlu(sndx,lt)/(1000._r8 + 3._r8*rlu(sndx,lt))
+                   endif
+                endif
 
-             end if
-          end do
-       end if
-    end do
+             endif
+          enddo
+       endif
+    enddo
 
-    species_loop2 : do ispec = 1,gas_pcnst
-       m = map_dvel(ispec)
+    do ispec = 1,gas_pcnst
+       idx_drydep = map_dvel(ispec)
        if( has_dvel(ispec) ) then
-          if( ispec /= o3_ndx .and. ispec /= o3a_ndx .and. ispec /= so2_ndx ) then
+          if( ispec /= so2_ndx ) then
              do lt = beglt,endlt
-                if( lt /= 7 ) then
-                   do i = 1,ncol
-                      if( fr_lnduse(i,lt) ) then
+                if( lt /= lt_for_water ) then
+                   do icol = 1,ncol
+                      if( fr_lnduse(icol,lt) ) then
                          !-------------------------------------------------------------------------------------
                          ! no effect if sfc_temp < O C
                          !-------------------------------------------------------------------------------------
-                         if( sfc_temp(i) > tmelt ) then
-                            if( has_dew(i) ) then
-                               rlux(i,lt,ispec) = 1._r8/((1._r8/(3._r8*rlux(i,lt,ispec))) &
-                                    + 1.e-7_r8*heff(i,m) + foxd(m)/rlux_o3(i,lt))
-                            end if
-                         end if
-
-                      end if
-                   end do
-                end if
-             end do
+                         if( sfc_temp(icol) > tmelt ) then
```

Comment:
same comment to merge the three if conditions into one

Reply:
merged